src/scripts/branch_review.py

```diff
@@ -0,0 +1,239 @@
+    if quiet:
+        logger.setLevel(level=logging.ERROR)
+
+
+@main.command()
+@click.option("-b", "--branch-id", help="Branch ID")
+@click.option("-B", "--branch-id-file", help="TSV with one column of obsoletion CURIEs")
+@click.option(
+    "-f",
+    "--obsoletion-candidates-file",
+    help="TSV with one column of obsoletion CURIEs",
+)
+@click.option("-o", "--output-file", help="Path to report file")
+def create_review_table(
+    branch_id, branch_id_file, obsoletion_candidates_file, output_file
+):
+    # Open the TSV file and read it into a list
+    with open(obsoletion_candidates_file, "r") as f:
+        reader = csv.reader(f, delimiter="\t")
+        obsoletion_candidates = [row[0] for row in reader]
+
+    if branch_id is None:
+        if branch_id_file:
+            with open(branch_id_file, "r") as f:
+                reader = csv.reader(f, delimiter="\t")
+                branch_ids = [row[0] for row in reader]
+        else:
+            raise (
+                ValueError(
+                    "Please provide either `--branch-id` OR `--branch-id-file` to get results."
+                )
+            )
+    else:
+        branch_ids = [branch_id]
+
+    branch_descendants_dict = {
+        id: list(OI.descendants(start_curies=id, predicates=[IS_A, PART_OF]))
+        for id in branch_ids
+    }
+    obsoletion_candidates_relationships = list(
+        OI.relationships(subjects=obsoletion_candidates, predicates=[IS_A, PART_OF])
+    )
+    column_names = [
+        "BranchID",
+        "AffectedDisease",
+        "AffectedDiseaseLabel",
+        "AffectedDiseaseDefinition",
+        "ObsoleteParents",
+        "ParentInsideBranch",
+        "ParentsOutsideBranch",
+        "AncestorsInsideBranch",
+        "AncestorsOutsideBranch",
+        "AffectedStatus",
+    ]
+    with open(output_file, "w", newline="") as csvfile:
+        writer = csv.writer(csvfile, delimiter="\t")
+        writer.writerow(column_names)  # writing the headers
+
+        for branch_id in branch_ids:
+            # all classes of obsoletion_candidate_list that are descendants of branch_id
+            obsoletion_candidates_branch = set(
+                branch_descendants_dict[branch_id]
+            ).intersection(set(obsoletion_candidates))
+            obsoletion_candidates_branch_relationships = list(
+                OI.relationships(
+                    subjects=obsoletion_candidates_branch, predicates=[IS_A, PART_OF]
+                )
+            )
+
+            affected_mondo_ids = list(
+                chain.from_iterable(
+                    get_children_from_relations(
+                        curie=id,
+                        relationships=obsoletion_candidates_branch_relationships,
+                    )
+                    for id in obsoletion_candidates_branch
+                )
+            )
+
+            if len(affected_mondo_ids) > 0:
+                for obsoletion_candidate in affected_mondo_ids:
```

Comment:
```suggestion
                for affected_id in affected_mondo_ids:
```

Best rename this variable as it makes the rest of the code confusing to read.